library/src/scala/quoted/Quotes.scala

```diff
@@ -3687,6 +3696,15 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
       */
       def newVal(parent: Symbol, name: String, tpe: TypeRepr, flags: Flags, privateWithin: Symbol): Symbol
 
+      /** Generates a new `val` or `lazy val` symbol that overrides another definition.
```

Comment:
No. This design does not compose well. It might be better to have an API to compute the methodic type of the override signature and use the other `newVal` overloads to create the symbol with the appropriate flags. This would require the introduction of `.info`.